pages/favourites.js

```diff
@@ -0,0 +1,46 @@
+import ActivityList, { useActivities } from "@/components/ActivitiesList";
+import BackButton from "@/components/BackButton";
+import { useFavourites } from "@/hooks/useFavourites";
+
+export default function FavouritesPage() {
+  const { favourites, getIsFavourite } = useFavourites();
+  const { activities, error, isLoading } = useActivities();
+
+  if (!favourites) {
+    return null;
+  }
+
+  if (isLoading) {
+    return <h1>Loading...</h1>;
+  }
+
+  if (error) {
+    return <h1>{error.message}</h1>;
+  }
+
+  if (!activities) {
+    return <h1>Activities not found</h1>;
+  }
+
+  const favouriteActivities = activities.filter((activity) =>
+    getIsFavourite(activity._id)
+  );
+
+  console.log(favouriteActivities);
+
+  if (favouriteActivities.length === 0) {
+    return <h1>No favourites yet</h1>;
+  }
```

Comment:
In the case of no favorites you display two 'h1' elements. 
Also the Back Button is not visible in this case.
Instead you should better render in the JSX either the 'ActivityList' or the 'No favorites' message.